### ISSUE TITLE
Curl example does not respect non-JSON encType when using POST

### ISSUE THREAD

**Issue by duckpunch** [interagent/prmd#296]
I think the greater issue is that `encType` for links is just not supported, but this is the specific problem I ran into.

In the [Hyper-Schema spec](http://json-schema.org/latest/json-schema-hypermedia.html#anchor37), there is a snippet that looks like this.

```json
{
    "links": [{
        "encType": "application/x-www-form-urlencoded",
        "method": "GET",
        "href": "/Product/",
        "properties": {
            "name": {
                "description": "name of the product"
            }
        }
    }]
}
```

In my case, I'm essentially trying the same thing, but instead with `method=POST`.  The curl example from the outputted Markdown shows a JSON body.

**Comment by geemus** [comment]
I think this was just a naive, good-enough solution since we only happen to use json bodies. I'd be open to discussing a PR that uses this info to make choices about how to format the examples. I suspect we should just maintain json as the default for backwards compatibility.

**Comment by duckpunch** [comment]
How about switching the example when either `properties` or `schema` is provided for a link?

That is, if `schema` is provided, it works like it does today.

If `properties` is provided, the example shows

```
curl <url>
    -d foo=bar \
    -d bar=foo \
    -H "Content-Type: <encType>"
```

If both are provided, prefer using schema (maintaining the current implementation).

Alternatively, there could be an option to either override or omit the curl example.

**Comment by geemus** [comment]
Hmm. I worry that leaves a lot to implied behavior, could we just explicitly check/rely upon the contents of encType instead of presence/absence of other data?

**Comment by duckpunch** [comment]
Switching on the `encType` seems reasonable.

I'm not quite sure what to do if you say something like `encType=application/x-www-form-urlencoded` and provide a `schema`.  The [spec](http://json-schema.org/latest/json-schema-hypermedia.html#anchor38) almost seems to suggest that it's a JSON body.

> for a POST request, this would define the body

**Comment by geemus** [comment]
Yeah, maybe we could just do things if encType and example are both present? But maybe it should try to do something with the schema. Yeah, the spec does point to GET query, POST body, but I think you can also put form encoded stuff in the body and be valid?
